### DISCUSSION TITLE
Is it possible to run fast-check continuously as a proxy for a fuzzer?

### DISCUSSION THREAD

**Discussion by jasikpark** [dubzzz/fast-check#2769] [Q&A] [answered]
I'm looking into how to fuzz typescript simply and effectively - I've found https://gitlab.com/gitlab-org/security-products/analyzers/fuzzers/jsfuzz and used it successfully with a manually created commonjs script, but I've had trouble integrating it with an existing ts project.

`fast-check` was simple to set up and integrate without testing setup, but it would be nice to allow for continuous exploration of the input space for properties rather than the set runtime that fast-check runs with

**Comment by dubzzz** [comment, marked as answer]
Not sure to fully answer your question, but if what you are looking for is to use fast-check to run a property against an infinite number of items then you could pass it an option called `numRuns`.

Based on what I read on the documentation you attached to the discussion, I believe fast-check will be able to do quite a great job in many scenarios covered by fuzzers. But it will not be coverage-based (generated values, except for shrink case, are fully independent from each others) and will not be able to detect infinite loop in sync code (not implemented yet).

**Reply by jasikpark** [reply]
Ok, I'll look into `numRuns`! I'm trying to explore how to setup something similar to:

- `go fuzz`: https://go.dev/doc/tutorial/fuzz
- `cargo fuzz`: https://rust-fuzz.github.io/book/introduction.html

which is certainly separate from prop-testing in the sense that it's intended to be a continually running process, exploring the input space, versus prop-testing which is more inline with traditional unit testing CI and should resolve in a timely manner.

the thing I appreciate about fast-check is that it is import based, it works with esm / typescript easily, and has good extensive documentation.

using `jsfuzz` with something other than a hand-written commonjs module has been troublesome.

Thanks for the response!
